diff --git a/node_modules/react-native-smooth-pincode-input/src/index.js b/node_modules/react-native-smooth-pincode-input/src/index.js
index 0a00f22..961b184 100644
--- a/node_modules/react-native-smooth-pincode-input/src/index.js
+++ b/node_modules/react-native-smooth-pincode-input/src/index.js
@@ -6,8 +6,9 @@ import {
   TextInput,
   StyleSheet,
   I18nManager,
-  ViewPropTypes,
 } from 'react-native';
+import ViewPropTypes from 'deprecated-react-native-prop-types/DeprecatedViewPropTypes';
+import TextStylePropTypes from 'deprecated-react-native-prop-types/DeprecatedTextStylePropTypes'
 import * as Animatable from 'react-native-animatable';
 
 const styles = StyleSheet.create({
@@ -292,8 +293,8 @@ SmoothPinCodeInput.propTypes = {
   cellStyleFocused: ViewPropTypes.style,
   cellStyleFilled: ViewPropTypes.style,
 
-  textStyle: Text.propTypes.style,
-  textStyleFocused: Text.propTypes.style,
+  textStyle: TextStylePropTypes,
+  textStyleFocused: TextStylePropTypes,
 
   animated: PropTypes.bool,
   animationFocused: PropTypes.oneOfType([
